lib/db/backend/badger_backend.go

```diff
@@ -0,0 +1,377 @@
+// Copyright (C) 2019 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package backend
+
+import (
+	"bytes"
+	"sync"
+
+	badger "github.com/dgraph-io/badger/v2"
+)
+
+const (
+	checkpointFlushMinSize = 1 << 20
+)
+
+func OpenBadger(path string) (Backend, error) {
+	opts := badger.DefaultOptions(path + ".badger") // temporary path safety
+	opts = opts.WithMaxCacheSize(64 << MiB).WithCompactL0OnClose(false)
+	opts.Logger = nil
```

Comment:
IIRC it logged some stats and stuff to stdout by default. Until proven otherwise there's nothing I want to hear from the db library that it doesn't tell me via an `error` return so I dropped it. :)